wrangler-core/src/main/java/io/cdap/wrangler/utils/ColumnConverter.java

```diff
@@ -28,26 +28,47 @@ private ColumnConverter() {
   }
 
   /**
-   * Renames a column.
+   * Renames a column. This method assumes the column name is case insensitive
    *
    * @param row source record to be modified.
    * @param column name of the column within source record.
    * @param toName the target name of the column.
-   * @throws DirectiveExecutionException when a column matching the target name already exists.
+   * @throws DirectiveExecutionException when a column matching the target name already exists or the given column
+   * does not exist
    */
   public static void rename(String directiveName, Row row, String column, String toName)
     throws DirectiveExecutionException {
-    int idx = row.find(column);
-    int existingColumn = row.find(toName);
-    if (idx != -1) {
-      if (existingColumn == -1) {
-        row.setColumn(idx, toName);
-      } else {
-        throw new DirectiveExecutionException(
-          directiveName, String.format("Column '%s' already exists. Apply the 'drop %s' directive before " +
-                                         "renaming '%s' to '%s'.",
-                                       toName, toName, column, toName));
-      }
+    rename(directiveName, row, column, toName, false);
+  }
+
+  /**
+   * Renames a column.
+   *
+   * @param row source record to be modified.
+   * @param column name of the column within source record.
+   * @param toName the target name of the column.
+   * @param exactMatch true for case sensitive col name, false for case insensitive
+   * @throws DirectiveExecutionException when a column matching the target name already exists or the given column
+   * does not exist
+   */
+  public static void rename(String directiveName, Row row, String column, String toName, boolean exactMatch)
+    throws DirectiveExecutionException {
+    int idx = exactMatch ? row.findExact(column) : row.find(column);
```

Comment:
this is also backwards incompatible. The checks should all be case insensitive, with the exception of if before and after are just different casing.

Reply:
In other words, if there is a row <a:0, b:1>

rename A a2 // should not fail, result in <a2:0, b:1>
rename a B // should behave the same (failure)
rename a A // should no longer fail, result in <A:0, b:1>